### PR TITLE
Add UMI.CMS detection

### PR DESCRIPTION
**Pull request by AlexByte**
Sites for test

- [first](http://key.ru)
- [second](http://3dmars.ru)
- [third](http://rfdyn.com)
- [fourth](http://samges.ru)

### CODE REVIEW

File: src/apps.json [hunk 1]
```diff
@@ -7890,6 +7890,18 @@
 			"url": "/umbraco/login\\.aspx(?:$|\\?)",
 			"website": "umbraco.com"
 		},
+		"UMI.CMS": {
+			"cats": [
+				1
+			],
+			"headers": {
+				"X-Generated-By": "",
```
Comment (jvoisin) [inline src/apps.json]: This should be `"X-Generated-By": "UMI.CMS",`

File: src/apps.json [hunk 2]
```diff
@@ -7890,6 +7890,18 @@
...
+				"X-CMS-Version": "(.+)\\;version:\\1"
```
Comment (jvoisin) [inline src/apps.json]: This should be `"X-CMS-Version": "([0-9.])\\;version:\\1"`

File: src/apps.json (hunk 1 above)
Comment (AlexByte) [inline src/apps.json]: Its unique header

File: src/apps.json (hunk 2 above)
Comment (AlexByte) [inline src/apps.json]: This header contains only the version number

File: src/apps.json (hunk 1 above)
Comment (jvoisin) [inline src/apps.json]: [No](https://docs.seneca.nl/smartsite-doc.net?id=31264 ) [it's]( https://bugs.launchpad.net/launchpad/+bug/82669 ) [not]( http://rawdc.org/web/tfmail.html ).

File: src/apps.json (hunk 2 above)
Comment (jvoisin) [inline src/apps.json]: Sure, but tightening the regular expression doesn't hurt.

File: src/apps.json (hunk 1 above)
Comment (AlexByte) [inline src/apps.json]: fixed

File: src/apps.json (hunk 2 above)
Comment (AlexByte) [inline src/apps.json]: fyi ^([0-9.])$ is wrong pattern
https://regex101.com/r/OUOXZr/1

Comment (jvoisin) [inline src/apps.json]: Yup, I forgot a `+`, sorry :/

**Comment by AlexByte** [comment]
Header `X-CMS-Version` is used in [Vakarm CMS](https://samesites.net/web-technology/vakarm-cms/1), so version of UMI.CMS cannot be displayed in extension. Need to fix

**Comment by jvoisin** [comment]
Good catch :)

**Comment by AlexByte** [comment]
@jvoisin There is no possibility in extension to load version number from another header (if header `X-Generated-By` exist) instead of adding another header detector?

**Comment by AliasIO** [comment]
Wappalyzer will still display the application if the confidence is set to zero. This will result in false positives.

**Comment by AlexByte** [comment]
@AliasIO I guess it would useful not to show detections with zero confidence.

**Comment by AliasIO** [comment]
Agree, possibly even anything under 50%.

**Comment by AlexByte** [comment]
@AliasIO when you plan not to show detections with zero confidence?

**Comment by jvoisin** [comment]
Why would you want to detect something with a zero percentage of chance of actually being used?

**Comment by AlexByte** [comment]
@jvoisin because of [this](https://github.com/AliasIO/Wappalyzer/pull/1390#issuecomment-282476611). I want to get additional version info from header, but don't to detect by header.

**Comment by jvoisin** [comment]
I don't think that it's possible.

**Comment by AlexByte** [comment]
@jvoisin Not possible what?

**Comment by jvoisin** [comment]
To have conditions like the one you're looking for.

**Comment by AlexByte** [comment]
@jvoisin version info is loaded from another header, but it can cause false positive detection with zero confidence, whats why detections with zero confidence should not be visible.

**Comment by AliasIO** [comment]
Do you want to remove the `X-CMS-Version` header pattern?

**Comment by gadcam** [comment]
@AliasIO couldn't we implement the zero confidence trick ? As it @AlexByte could keep its <code>X-CMS-Version</code> header pattern.

**Comment by AlexByte** [comment]
@AliasIO i removed `X-CMS-Version` header pattern.

**Comment by AliasIO** [comment]
Thanks. I also finally added the no confidence check.

**Pull request merged**